### PR TITLE
Add groupkey in init msg

### PR DESCRIPTION
**Pull request by Powersource**
for https://github.com/ssbc/ssb-tribes2/issues/40

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "fast-deep-equal": "^3.1.3",
     "lodash.get": "^4.4.2",
     "lodash.set": "^4.3.2",
-    "private-group-spec": "^3.0.0",
+    "private-group-spec": "github:ssbc/private-group-spec#groupkey-in-init",
```
Comment (Powersource) [inline package.json:42]: https://github.com/ssbc/private-group-spec/pull/25

File: index.js [hunk 2]
```diff
@@ -143,30 +143,31 @@ module.exports = {
...
-        },
-      }
-      if (!initSpec(content)) return cb(new Error(initSpec.errorsString))
-
       findOrCreateGroupFeed(null, function gotGroupFeed(err, groupFeed) {
         // prettier-ignore
         if (err) return cb(clarify(err, 'Failed to find or create group feed when creating a group'))
 
-        const secret = secretKeyFromPurpose(groupFeed.purpose)
+        const secret = secretKeyFromString(groupFeed.purpose)
```
Comment (staltz) [inline index.js:155]: Not a big deal, but a small suggestion:

```suggestion
        const groupKey = secretKeyFromString(groupFeed.purpose)
```

We seem to be going back and forth with this naming, "secret" or "group key", but we could try to be at least internally consistent (all namings of this thing in ssb-tribes2 are the same).

File: index.js [hunk 3]
```diff
@@ -257,37 +258,37 @@ module.exports = {
...
 
-            return cb(null, data)
-          })
+        const data = {
+          id: buildGroupId({
+            groupInitMsg,
+            groupKey: secret.toBuffer(),
+          }),
+          secret: secret.toBuffer(),
```
Comment (staltz) [inline index.js:274]: Similarly, could name this `groupKey`

File: index.js [hunk 4]
```diff
@@ -257,37 +258,37 @@ module.exports = {
...
+          root: fromMessageSigil(groupInitMsg.key),
+          subfeed: groupFeed.keys,
         }
-      )
+
+        ssb.box2.addGroupInfo(data.id, {
+          key: data.secret,
```
Comment (staltz) [inline index.js:280]: Case in point: the key is the secret is the key is the secret.

**Comment by staltz** [comment]
Tests need to be updated, they are failing just because the new field `groupKey` was unexpected.

File: index.js (hunk 2 above)
Comment (Powersource) [reply index.js:155]: Problem here is the spec calls it groupKey and the tribes2 api says secret. I think the proper way for fixing is a separate issue/pr.

Comment (Powersource) [reply index.js:155]: this pr is just for conforming to the spec

Comment (staltz) [reply index.js:155]: Okay!

File: package.json (hunk 1 above)
Comment (staltz) [reply package.json:42]: After merging that one, and publishing, then we can merge this tribes2 PR

**Review by staltz — APPROVED** [review]
LGTM!

**Pull request merged**